Observer/BackfillTransactions.php

```diff
@@ -15,309 +15,347 @@
         \Magento\Framework\Bulk\BulkManagementInterface $bulkManagement,
         \Magento\AsynchronousOperations\Api\Data\OperationInterfaceFactory $operationFactory,
         \Magento\Framework\DataObject\IdentityGeneratorInterface $identityService,
         \Magento\Framework\Serialize\SerializerInterface $serializer,
-        \Magento\Authorization\Model\UserContextInterface $userContext
+        \Magento\Authorization\Model\UserContextInterface $userContext,
+        \Taxjar\SalesTax\Model\Configuration $taxjarConfig
     ) {
         $this->request = $request;
-        $this->storeManager = $storeManager;
-        $this->logger = $logger->setFilename(TaxjarConfig::TAXJAR_TRANSACTIONS_LOG)->force();
+        $this->logger = $logger;
         $this->orderRepository = $orderRepository;
-        $this->filterBuilder = $filterBuilder;
+        $this->storeManager = $storeManager;
         $this->searchCriteriaBuilder = $searchCriteriaBuilder;
-        $this->taxjarConfig = $taxjarConfig;
         $this->bulkManagement = $bulkManagement;
         $this->operationFactory = $operationFactory;
-        $this->identityService = $identityService;
         $this->serializer = $serializer;
         $this->userContext = $userContext;
+        $this->taxjarConfig = $taxjarConfig;
+        $this->uuid = $identityService->generateId();
+
+        $this->logger->setFilename(\Taxjar\SalesTax\Model\Configuration::TAXJAR_TRANSACTIONS_LOG)->force();
     }
 
     /**
-     * @param Observer $observer
-     * @throws LocalizedException
+     * @throws \Magento\Framework\Exception\LocalizedException|\Exception|\Throwable
      */
-    public function execute(Observer $observer): void
+    public function execute(\Magento\Framework\Event\Observer $observer)
     {
-        $this->apiKey = $this->taxjarConfig->getApiKey();
+        $this->observer = $observer;
 
-        if (!$this->apiKey) {
-            throw new LocalizedException(
-                __('Could not sync transactions with TaxJar. Please make sure you have an API key.')
-            );
+        try {
+            if (!$this->taxjarConfig->getApiKey()) {
+                throw new \Magento\Framework\Exception\LocalizedException(
+                    __('Could not sync transactions with TaxJar. Please make sure you have an API key.')
+                );
+            }
+
+            $criteria = $this->getSearchCriteria();
+            $orders = $this->getOrders($criteria, (bool)$this->getInput('force'));
+
+            if (!empty($orders)) {
+                $this->syncTransactions($orders);
+            }
+
+            $this->success();
+        } catch (\Exception $e) {
+            $this->fail($e);
         }
+    }
 
-        $this->logger->log(__('Initializing TaxJar transaction sync'));
+    /**
+     * @throws \Magento\Framework\Exception\LocalizedException
+     */
+    public function success()
+    {
+        $this->log('Transaction sync successfully scheduled.');
```

Comment:
What would you think about including the number of transactions we scheduled to sync. So for example say we found 5 orders it would be something like:

"Successfully scheduled transaction sync for 5 orders and their credit memos".

Then if we don't find any orders that need to be synced it could be something like:

"No orders were found that need to be synced to TaxJar".

The messaging might need a little work but in general I think it might make it more clear to the merchant what is happening.